cmd/tools/asup/asup.go

```diff
@@ -0,0 +1,441 @@
+		if err = os.MkdirAll(payloadDir, 0777); err != nil {
+			return "", fmt.Errorf("could not create asup payload directory %s: %w", payloadDir, err)
+		}
+	}
+
+	return path.Join(payloadDir, fmt.Sprintf("%s_%s", pollerName, "payload.json")), nil
+}
+
+func getClusterNodeInfo(client *client.Client) ([]string, error) {
+	var (
+		response *node.Node
+		nodes    []*node.Node
+		err      error
+		uuids    []string
+	)
+
+	request := "cluster-node-get-iter"
+
+	if response, err = client.InvokeRequestString(request); err != nil {
+		return nil, fmt.Errorf("failure invoking zapi: %s %w", request, err)
+	}
+
+	if attrs := response.GetChildS("attributes-list"); attrs != nil {
+		nodes = attrs.GetChildren()
+	}
+
+	for _, n := range nodes {
+		uuid := n.GetChildContentS("node-uuid")
+		uuids = append(uuids, uuid)
+	}
+
+	return uuids, nil
+}
+
+func (target *targetInfo) getClusterInfo(connection *client.Client) error {
+	var (
+		response *node.Node
+		err      error
+	)
+
+	isClustered := target.Model == "cdot"
+	// fetch system name and serial number
+	request := "cluster-identity-get"
+	if !isClustered {
+		request = "system-get-info"
+	}
+
+	if response, err = connection.InvokeRequestString(request); err != nil {
+		return fmt.Errorf("failure invoking zapi: %s %w", request, err)
+	}
+
+	if isClustered {
+		if attrs := response.GetChildS("attributes"); attrs != nil {
+			if info := attrs.GetChildS("cluster-identity-info"); info != nil {
+				target.ClusterUuid = info.GetChildContentS("cluster-uuid")
+			}
+		}
+	} else {
+		if info := response.GetChildS("system-info"); info != nil {
+			// There is no uuid for non cluster mode, using system-id.
+			target.ClusterUuid = info.GetChildContentS("system-id")
+		}
+	}
+	return nil
+}
+
+func (node *NodeInstanceInfo) getNodeInfo(md *matrix.Matrix, connection *client.Client) error {
+	var uuids []string
+	var err error
+	if uuids, err = getClusterNodeInfo(connection); err != nil {
+		fmt.Printf("issue while fetching node uuids")
+	}
+	node.Count, _ = md.LazyGetValueInt64("count", "instance")
+	node.DataPoints, _ = md.LazyGetValueInt64("count", "data")
+	node.PollTime, _ = md.LazyGetValueInt64("poll_time", "data")
+	node.ApiTime, _ = md.LazyGetValueInt64("api_time", "data")
+	node.ParseTime, _ = md.LazyGetValueInt64("parse_time", "data")
+	node.PluginTime, _ = md.LazyGetValueInt64("plugin_time", "data")
+	node.NodeUuid = uuids
+	return nil
+}
+
+func getdata(config, pollername string) *client.Client {
+	var (
+		params     *node.Node
+		err        error
+		connection *client.Client
+	)
+
+	if params, err = conf.GetPoller(config, pollername); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	if connection, err = client.New(params); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	if err = connection.Init(2); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	return connection
+}
+
+func getSha1Uuid(clusterUuid string) string {
+	shaHash := sha1.New()
```

Comment:
why are we even doing sha1 on this. This value is anyways included in payload.

Reply:
if we don't cluster-uuid and harvest-uuid will be in conflict